service/grpc/codelingo/codelingo.pb.go

```diff
@@ -258,13 +243,188 @@ func (m *GenFact) GetFactName() string {
 	return ""
 }
 
-func (m *GenFact) GetProperties() map[string]string {
+func (m *GenFact) GetProperties() map[string]*GenProperty {
 	if m != nil {
 		return m.Properties
 	}
 	return nil
 }
 
+func (m *GenFact) GetChildren() []*GenFact {
+	if m != nil {
+		return m.Children
+	}
+	return nil
+}
+
+type GenProperty struct {
+	// Types that are valid to be assigned to Value:
+	//	*GenProperty_String_
+	//	*GenProperty_Int
+	//	*GenProperty_Float
+	//	*GenProperty_Bool
+	Value isGenProperty_Value `protobuf_oneof:"value"`
+}
+
+func (m *GenProperty) Reset()                    { *m = GenProperty{} }
+func (m *GenProperty) String() string            { return proto.CompactTextString(m) }
+func (*GenProperty) ProtoMessage()               {}
+func (*GenProperty) Descriptor() ([]byte, []int) { return fileDescriptor0, []int{6} }
+
+type isGenProperty_Value interface {
+	isGenProperty_Value()
+}
+
+type GenProperty_String_ struct {
+	String_ string `protobuf:"bytes,1,opt,name=string,oneof"`
+}
+type GenProperty_Int struct {
+	Int int64 `protobuf:"varint,2,opt,name=int,oneof"`
+}
+type GenProperty_Float struct {
+	Float float32 `protobuf:"fixed32,3,opt,name=float,oneof"`
+}
+type GenProperty_Bool struct {
+	Bool bool `protobuf:"varint,4,opt,name=bool,oneof"`
+}
+
+func (*GenProperty_String_) isGenProperty_Value() {}
+func (*GenProperty_Int) isGenProperty_Value()     {}
+func (*GenProperty_Float) isGenProperty_Value()   {}
+func (*GenProperty_Bool) isGenProperty_Value()    {}
+
+func (m *GenProperty) GetValue() isGenProperty_Value {
+	if m != nil {
+		return m.Value
+	}
+	return nil
+}
+
+func (m *GenProperty) GetString_() string {
+	if x, ok := m.GetValue().(*GenProperty_String_); ok {
+		return x.String_
+	}
+	return ""
+}
+
+func (m *GenProperty) GetInt() int64 {
+	if x, ok := m.GetValue().(*GenProperty_Int); ok {
+		return x.Int
+	}
+	return 0
+}
+
+func (m *GenProperty) GetFloat() float32 {
+	if x, ok := m.GetValue().(*GenProperty_Float); ok {
+		return x.Float
+	}
+	return 0
+}
+
+func (m *GenProperty) GetBool() bool {
+	if x, ok := m.GetValue().(*GenProperty_Bool); ok {
+		return x.Bool
+	}
+	return false
+}
+
+// XXX_OneofFuncs is for the internal use of the proto package.
+func (*GenProperty) XXX_OneofFuncs() (func(msg proto.Message, b *proto.Buffer) error, func(msg proto.Message, tag, wire int, b *proto.Buffer) (bool, error), func(msg proto.Message) (n int), []interface{}) {
```

Comment:
don't use underscores in Go names; method XXX_OneofFuncs should be XXXOneofFuncs